### ISSUE TITLE
Add `declaration-empty-line-before` option

### ISSUE THREAD

**Issue by Volker-E** [wikimedia/stylelint-config-wikimedia#73]
Propose to add `declaration-empty-line-before` and orientate on stylelint-config-standard here, with exception, that we shouldn't need to ignore `inside-single-line-block`:

```lang=json
"declaration-empty-line-before": [  "never", {
    ignore: [
        "after-comment",
        "inside-single-line-block",
    ],
} ],
```
Needs some closer testing, I don't want to see new lines except for rare exceptions.

That will do away with messy, inconsistent sometimes  property sequences like:
```
.mw-notification {
	padding: 0.75em 1.5em;
	margin-bottom: 0.5em;
	border: solid 1px #ddd;
	background-color: #fff;
	/* Click handler in mediawiki.notification.js */
	cursor: pointer;

	opacity: 0;
	-webkit-transform: translateX( 35px );
	transform: translateX( 35px );
	-webkit-transition: opacity 0.35s ease-in-out, -webkit-transform 0.35s ease-in-out;
	transition: opacity 0.35s ease-in-out, transform 0.35s ease-in-out;
}
```

**Comment by Volker-E** [comment]
Neither OOUI, nor core does have false positives or problematic flags with that setting…

**Comment by jdforrester** [comment]
Sounds good to me.
